NonEmptyList/fromList

```diff
@@ -0,0 +1,16 @@
+  λ(a : Type)
+→ λ(list : List a)
+→ List/fold
+  a
+  list
+  (Optional (./Type a))
```

Comment:
One minor nit: Since we use `./Type` in multiple places could we use a let binding:
```haskell
let NonEmptyList = ./Type
in ...
```

The reason for this is if we start to use freezing then we won't have multiple SHA diffs.

Reply:
I followed the convention of naming it `NonEmpty`. I hope that's okay. Happy to change it to `NonEmptyList`, if not.

Reply:
I think `NonEmptyList` would be better since that's the convention we've gone for in the directory naming :)